src/App.tsx

```diff
@@ -8,7 +8,66 @@ import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+
+type FilterParams = {
+  filterBy: string;
+  query: string;
+};
+
+function getFilteredTodos(todos: Todo[], { filterBy, query }: FilterParams) {
+  let filteredTodos = [...todos];
+  const preparedQuery = query.trim().toLocaleLowerCase();
+
+  switch (filterBy) {
+    case 'active':
+      filteredTodos = todos.filter(todo => !todo.completed);
+      break;
+
+    case 'completed':
+      filteredTodos = todos.filter(todo => todo.completed);
+      break;
+  }
+
+  if (preparedQuery) {
+    return filteredTodos.filter(todo =>
+      todo.title.toLowerCase().includes(preparedQuery),
+    );
+  }
+
+  return filteredTodos;
+}
+
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [loading, setLoading] = useState(false);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+
+  const [filterBy, setFilterBy] = useState('all');
+  const [query, setQuery] = useState('');
+
+  const filteredTodos = getFilteredTodos(todos, { filterBy, query });
+
+  useEffect(() => {
+    setLoading(true);
```

Comment:
Instead of `setLoading(true)` as soon as page renders (in useEffect), use `true` as `loading`'s initial state;

const [loading, setLoading] = useState(true);